tests/publiccloud/check_registercloudguest.pm

```diff
@@ -19,7 +19,9 @@ use strict;
 use utils;
 use publiccloud::utils;
 use publiccloud::ssh_interactive 'select_host_console';
-use version_utils 'is_sle';
+
+my $path = is_sle('=15-SP2') ? '/usr/sbin/' : '';    # 15-SP2 is the oldest version that needs fullpaths
+my $regcode_param = (is_byos()) ? "-r " . get_required_var('SCC_REGCODE') : '';
```

Comment:
Some notes, please:

1. These 2 lines of code are already performed in `sub registercloudguest`:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/d9fcc37e50f397fc08bd5a60ac0d3d6e05ec067e/lib/publiccloud/utils.pm#L110: those should be removed from here and function calls adapted to that.
- We should eventually change code in this test, accordingly.

2. Moreover, by using this contruction, when **NOT BYOS** all parameters are `empty`, but in the function calls it still is executed without parameters, like `sudo registercloudguest`: but `SCC_REGCODE` is mandatory in that routine, then would fail when missing.
- Please consider that in your **new routines calls**, you should check `is_byos` too.

Reply:
@asmorodskyi wdyt, please?

Reply:
- I agree that we must update https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/d9fcc37e50f397fc08bd5a60ac0d3d6e05ec067e/lib/publiccloud/utils.pm#L113 to keep same logic as here `my $path = is_sle('=15-SP2') ? '/usr/sbin/' : '';`
- I don't think we can/should force Hristo to do more refactoring if he decided to do some , also I would even prefer to keep it for different PR 
- (2) is expected behavior !

Reply:
about (2) yes expected, I was only suggesting to Hristo to check calls of his `...new_registration` routines be aware of the is_byos, because in L#111-L#117 don't.

Reply:
sorry I don't see how and where exactly such awareness will improve current code . maybe if you will give us some code snippet it would be more clear ?

Reply:
In  code lines L#111 to L#117 the 2 new routines calls don't check `is_byos` before to run: is this expected?